diff --git a/src/mod/applications/mod_callcenter/mod_callcenter.c b/src/mod/applications/mod_callcenter/mod_callcenter.c
index 656af70..ac29832 100644
--- a/src/mod/applications/mod_callcenter/mod_callcenter.c
+++ b/src/mod/applications/mod_callcenter/mod_callcenter.c
@@ -1832,19 +1832,6 @@ static void *SWITCH_THREAD_FUNC outbound_agent_thread_run(switch_thread_t *threa
			switch_event_fire(&event);
		}

-		/* Record session if record-template is provided */
-		if (h->record_template) {
-			char *expanded = switch_channel_expand_variables(member_channel, h->record_template);
-			switch_channel_set_variable(member_channel, "cc_record_filename", expanded);
-			switch_ivr_record_session(member_session, expanded, 0, NULL);
-			if (expanded != h->record_template) {
-				switch_safe_free(expanded);
-			}
-		}
-
-		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(member_session), SWITCH_LOG_DEBUG, "Agent %s answered \"%s\" <%s> from queue %s%s\n",
-				h->agent_name, h->member_cid_name, h->member_cid_number, h->queue_name, (h->record_template?" (Recorded)":""));
-
		if ((o_announce = switch_channel_get_variable(member_channel, "cc_outbound_announce"))) {
			playback_array(agent_session, o_announce);
		}
@@ -1908,6 +1895,19 @@ static void *SWITCH_THREAD_FUNC outbound_agent_thread_run(switch_thread_t *threa
			cc_tier_update("state", cc_tier_state2str(CC_TIER_STATE_ACTIVE_INBOUND), h->queue_name, h->agent_name);
			cc_agent_update("state", cc_agent_state2str(CC_AGENT_STATE_IN_A_QUEUE_CALL), h->agent_name);

+			/* Record session if record-template is provided */
+			if (h->record_template) {
+				char *expanded = switch_channel_expand_variables(member_channel, h->record_template);
+				switch_channel_set_variable(member_channel, "cc_record_filename", expanded);
+				switch_ivr_record_session(member_session, expanded, 0, NULL);
+				if (expanded != h->record_template) {
+					switch_safe_free(expanded);
+				}
+			}
+
+			switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(member_session), SWITCH_LOG_DEBUG, "Agent %s answered \"%s\" <%s> from queue %s%s\n",
+				h->agent_name, h->member_cid_name, h->member_cid_number, h->queue_name, (h->record_template?" (Recorded)":""));
+
		}
		/* Wait until the agent hangup.  This will quit also if the agent transfer the call */
		while(bridged && switch_channel_up(agent_channel) && globals.running) {
